libp2p/transport/tcp/tcp.py

```diff
@@ -1,64 +1,72 @@
-import asyncio
-from socket import socket
-import sys
-from typing import List
+import logging
+from typing import Awaitable, Callable, List, Sequence, Tuple
 
 from multiaddr import Multiaddr
+import trio
+from trio_typing import TaskStatus
 
+from libp2p.io.trio import TrioTCPStream
 from libp2p.network.connection.raw_connection import RawConnection
 from libp2p.network.connection.raw_connection_interface import IRawConnection
 from libp2p.transport.exceptions import OpenConnectionError
 from libp2p.transport.listener_interface import IListener
 from libp2p.transport.transport_interface import ITransport
 from libp2p.transport.typing import THandler
 
+logger = logging.getLogger("libp2p.transport.tcp")
+
 
 class TCPListener(IListener):
-    multiaddrs: List[Multiaddr]
-    server = None
+    listeners: List[trio.SocketListener]
 
     def __init__(self, handler_function: THandler) -> None:
-        self.multiaddrs = []
-        self.server = None
+        self.listeners = []
         self.handler = handler_function
 
-    async def listen(self, maddr: Multiaddr) -> bool:
+    # TODO: Get rid of `nursery`?
+    async def listen(self, maddr: Multiaddr, nursery: trio.Nursery) -> None:
         """
         put listener in listening mode and wait for incoming connections.
 
         :param maddr: maddr of peer
         :return: return True if successful
         """
-        self.server = await asyncio.start_server(
-            self.handler,
+
+        async def serve_tcp(
```

Comment:
What is the reason for this wrapper function?  Is it just for the logging statement?

Reply:
Yes, it's just for logging.

Reply:
Worth adding a comment saying this explicitly